### PR TITLE
Fixes issue #142 - Mocha and Istanbul failing to report accurate cove…

### PR DESCRIPTION
**Pull request by patnolanireland**
As per my comments on

https://github.com/yeoman/generator-node/issues/142

here is my humble fix.

### CODE REVIEW

**Comment by hemanth** [comment]
//cc @SBoudrias @sindresorhus

**Comment by twixie** [comment]
looks okay. Please merge.

File: generators/gulp/templates/gulpfile.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ var coveralls = require('gulp-coveralls');
 <% } -%>
 <% if (babel) { -%>
 var babel = require('gulp-babel');
+require('babel-core/register');
```
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Why `require('babel-core/register')` rather than `require('babel/register')`?

Comment (patnolanireland) [inline generators/gulp/templates/gulpfile.js]: Honestly I just moved the original text contained in the mocha config to a require statement at the head of the file

```
<% if (babel) { -%>
    .pipe(mocha({reporter: 'spec', require: ['babel-core/register']}))
    <% } else { -%>
    .pipe(mocha({reporter: 'spec'}))
    <% } %>
```
At this stage there is no compelling reason for the require('babel-core/register') over require('babel/register') as they are both essentially the same.  Happy to make the change or alternatively explicitly use the compiler option when configuring mocha with the babel option which makes the intention very clear as to what is happening e.g.

```
var babelCompiler = require('babel/register');

/* The mocha initialisation within the pipe */ 
mocha({
    compilers: {
        js: babelCompiler
     }
})
```

Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Maybe adding a comment explaining the the require is doing would help. I'm fine with just requiring it.

Comment (patnolanireland) [inline generators/gulp/templates/gulpfile.js]: I've updated with a comment and as it turns out the require to 'babel-core/register' is required as this is the published name of the module.

**Comment by SBoudrias** [comment]
Merged on master.

**Comment by hemanth** [comment]
Thanks @SBoudrias

**Pull request closed without merging**